src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -22,13 +23,62 @@ public sealed class TelemetryConfiguration : IDisposable
         private static TelemetryConfiguration active;
 
         private readonly SnapshottingList<ITelemetryInitializer> telemetryInitializers = new SnapshottingList<ITelemetryInitializer>();
+        private ITelemetryChannel telemetryChannel = null;
         private TelemetryProcessorChain telemetryProcessorChain;
         private string instrumentationKey = string.Empty;
         private bool disableTelemetry = false;
         private TelemetryProcessorChainBuilder builder;
         private SnapshottingList<IMetricProcessor> metricProcessors = new SnapshottingList<IMetricProcessor>();
 
         /// <summary>
+        /// Indicates if this instance has been disposed of.
+        /// </summary>
+        private bool isDisposed = false;
+
+        /// <summary>
+        /// Indicates if we created the telemetry channel and should therefore dispose of it.
+        /// </summary>
+        private bool shouldDisposeChannel = false;
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public TelemetryConfiguration() : this(string.Empty, new InMemoryChannel())
+        {
+            this.shouldDisposeChannel = true;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        /// <param name="instrumentationKey">The instrumentation key this configuration instance will provide.</param>
+        public TelemetryConfiguration(string instrumentationKey) : this(instrumentationKey, new InMemoryChannel())
+        {
```

Comment:
This constructor is creating the new InMemoryChannel() - so it should set shouldDIsposeChannel = true right?